driver-core/src/main/com/mongodb/client/model/search/SearchOperator.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.client.model.search;
+
+import com.mongodb.annotations.Evolving;
+import com.mongodb.client.model.Aggregates;
+import org.bson.BsonDocument;
+import org.bson.conversions.Bson;
+
+import static org.bson.assertions.Assertions.notNull;
+
+/**
+ * The core part of the {@link Aggregates#search(SearchOperator, SearchOptions) $search} pipeline stage of an aggregation pipeline.
+ *
+ * @mongodb.atlas.manual atlas-search/operators-and-collectors/#operators Search operators
+ * @since 4.6
+ */
+@Evolving
+public interface SearchOperator extends Bson {
+    /**
+     * Returns a {@link SearchOperator} that tests if the {@code path} exists in a document.
+     *
+     * @param path The path to test.
+     * @return The requested {@link SearchOperator}.
+     * @mongodb.atlas.manual atlas-search/exists/ exists operator
+     */
+    static ExistsSearchOperator exists(final FieldSearchPath path) {
```

Comment:
There are two reasons at least one operator built in the API is needed:

- To be used in tests for `Aggregates.search` and `SearchFacet`. We can just write tests in an uglier manner by using `SearchOperator.of`, but I would like not to do that.
- To be used in the documentation of the `SearchOperator.of` method: all such methods include examples of constructing functionally equivalent objects via factory methods and typed built-in builders, and via a generic API.

There seems to be no problem in having factory methods for only a subset of operators (even if that subset contains a single element) because generally this will be the case as Atlas search evolves: Atlas adds a new operator/collector/etc., the driver API still does not support it via factory methods, as time goes by, the driver API evolves to support the new Atlas search feature.

Reply:
Even though the API will inevitably fall out of date as time passes, releasing an initial version of it that supports only a single operator will be confusing, since at the time of the release it won't even have methods all the current ones.  But I'm ok leaving it as is for this PR.